lib/transactions-common.js

```diff
@@ -1426,8 +1426,10 @@ Transact.prototype._processTransaction = function (txid, description, items, con
   
   // First, need to iterate over the changes that are going to be made and make sure that,
   // if there are hard removes, the db version of the doc gets stored on the transaction
+  // We only do this step if the remove is not instant - otherwise the doc would already have
+  // been removed from the collection by this point.   
   _.each(items, function (item, index) {
-    if (item.action === "remove" && item.hardDelete) {
+    if (item.action === "remove" && item.hardDelete && !item.instant) {
       // Get the existing doc and store it in the transaction record
       // We overwrite the temporary version of the doc from an instant remove on the client
       // Because chances are that the whole document was not available on the client
```

Comment:
This comment suggests that the code was originally intended to save instant-removed documents.  Did this never work, as far as we know?  If that's the case, I guess the comment should be updated to state that having an incomplete document for instant removes from the client is an unsolved problem.